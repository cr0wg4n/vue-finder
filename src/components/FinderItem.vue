<template>
  <div class="item" :class="{ selected: selected }">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: "FinderItem",
  props: {
    selected: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style lang="less" scoped>
.item {
  padding: 10px;
  width: 250px;

  &.selected {
    background-color: lightgray;
  }
}
</style>
