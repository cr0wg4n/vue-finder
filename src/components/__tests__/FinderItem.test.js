import { mount } from "@vue/test-utils";
import FinderItem from "../FinderItem";

describe("Tree Utils", () => {
  it("should match snapshot", () => {
    const wrapper = mount(FinderItem);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should match snapshot if expanded", () => {
    const wrapper = mount(FinderItem, {
      propsData: {
        expanded: true
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
