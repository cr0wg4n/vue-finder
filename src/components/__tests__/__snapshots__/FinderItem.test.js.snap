// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree Utils should match snapshot 1`] = `
<div
  class="item"
/>
`;

exports[`Tree Utils should match snapshot if expanded 1`] = `
<div
  class="item expanded"
/>
`;
